packages/datadog-plugin-aws-sdk/src/base.js

```diff
@@ -0,0 +1,91 @@
+'use strict'
+
+const Tags = require('opentracing').Tags
+const analyticsSampler = require('../../dd-trace/src/analytics_sampler')
+const awsHelpers = require('./aws_helpers')
+
+function createWrapRequest (tracer, config) {
+  config = normalizeConfig(config)
+
+  return function wrapRequest (request) {
+    return function requestWithTrace (operation, params, cb) {
+      const serviceName = awsHelpers.normalizeServiceName(this)
+
+      const baseTags = {
+        [Tags.SPAN_KIND]: 'client',
+        'span.type': 'http',
+        'service.name': config.service || `${tracer._service}-aws`,
+        'aws.agent': 'js-aws-sdk',
+        'aws.operation': operation,
+        'aws.region': (request.httpRequest && request.httpRequest.region) || this.config.region,
+        'aws.service': serviceName,
+        'component': serviceName
+      }
+
+      const childOf = tracer.scope().active()
+      const span = tracer.startSpan('aws.http', {
+        childOf,
+        tags: baseTags
+      })
+
+      awsHelpers.addAwsRequestTags(span, request, operation, params, serviceName)
+
+      analyticsSampler.sample(span, config.analytics)
+
+      // https://github.com/awsdocs/aws-javascript-developer-guide-v2/blob/
```

Comment:
Why this comment?

Reply:
Same question for other similar comments.

Reply:
most of this was just to help with code review so everyone is familiar with what the args are , i can remove these "pointing to aws doc reference" type comments if its just messy and not helpful. I did add some code comments related to choices on _why_ the instrumentation is handling `with callback` and `without callback` differently, that i think are useful, but again if the conventions here is to keep the comments to a minimum , happy to remove.

Reply:
The idea is mostly to not pollute the code with unnecessary comments, otherwise you unconsciously start to ignore them, similar to apps with a lot of warnings. When there is a comment, it should be required to understand why the code works a specific way, but in this case I felt like we're just describing AWS general best practices which is not that helpful. But I also wanted to make sure I didn't miss something important that would make this comment really needed.